models/user_setting.go

```diff
@@ -0,0 +1,109 @@
+// Copyright 2021 The Gitea Authors. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+package models
+
+import (
+	"strings"
+
+	"xorm.io/builder"
+)
+
+// UserSetting is a key value store of user settings
+type UserSetting struct {
+	ID     int64  `xorm:"pk autoincr"`
+	UserID int64  `xorm:"index"`              // to load all of someone's settings
+	Key    string `xorm:"varchar(255) index"` // ensure key is always lowercase
+	Value  string `xorm:"text"`
+}
```

Comment:
If the intention is that there is only one value per key+userID pair we should add a uniqueness constraint:

```suggestion
type UserSetting struct {
	ID     int64  `xorm:"pk autoincr"`
	UserID int64  `xorm:"index unique(key_userid)"`                     // to load all of someone's settings
	Key    string `xorm:"varchar(255) index unique(key_userid)"` // ensure key is always lowercase
	Value  string `xorm:"text"`
}
```

Then you can use an UPSERT pattern to update the key below.